### PR TITLE
Generate `dev-requirements.txt` + improve readme for contributors wanting to run tests

### PR DESCRIPTION
**Pull request by TechnicalPirate**
**Problem** - Currently the `cx_freeze` project doesn't have a `requirements` file nor any dependency management solution - As usage of the product requires the `importlib_metadata` dependency and additional dependencies are required to contribute as a developer ( `nose` / `pygments` ) this can be a barrier to contributors. 

**Solution** - ~This PR contributes a basic `pipenv` [guide here](https://realpython.com/pipenv-guide/) setup~ This PR includes a `dev-requirements.txt` listing the dependencies required for developers - it also contributes a `Readme.md` for the `/test/` directory explaining how contributors can execute the tests locally

**TLDR**:
~* Adds `pipfile` and `pipfile.lock` for contributors who use `pipenv`~
* Adds `dev-requirements.txt` to the project root
* Adds `readme.md` to the tests directory
* Adds an exclude of [Pycharms](https://www.jetbrains.com/pycharm/promo/?source=google&medium=cpc&campaign=14127625862&gclid=CjwKCAjwybyJBhBwEiwAvz4G7-1MJ0reiDFYx7WWQHOwAJwId7lSbae4m9NqtY4iWd6APMDISbohnxoCN90QAvD_BwE) `.idea` folder to the `.gitignore`

*Note* : [Discussion here](https://github.com/marcelotduarte/cx_Freeze/discussions/1186)

### CODE REVIEW

**Comment by marcelotduarte** [comment]
@TechnicalPirate About this PR I've been thinking that the best would be if you create a dev-requirements.txt or test-requirements.txt file as I see in other projects (for example in pyzmq), so don't force the developer to use pipenv, and he/she can use whatever he/she wants.

**Comment by TechnicalPirate** [comment]
@marcelotduarte - as you wish, give me 5min to feed the cat and i'll spit out a `dev-requirements.txt` and update this PR :) 

Does that mean you'd want the `pipenv` gubbins *removing* though? As by leaving it we'd support `pipenv` users as well out the box :)

**Comment by marcelotduarte** [comment]
How will I explain to you. I don't pipenv normally, and don't plan to use it. But I use it in a CI test just to see if cx_Freeze runs some samples. Then I will run the tests you are implementing. With the requirements we can create a Pipfile....

**Comment by marcelotduarte** [comment]
> gubbins

I don't know this word nor google translator knows ;-)

**Comment by TechnicalPirate** [comment]
@marcelotduarte - Check now

I have:

* generated a `dev-requirements.txt` from my local venv
* Added `pipenv` specific files to `.gitignore`
* slightly updated the README.md in the test folder

Hope that helps!

_As for [gubbins](https://www.merriam-webster.com/dictionary/gubbins) - apologies for slang usage - i was referring to the files used by `pipenv` ... which have been removed ;) :D_

**Comment by marcelotduarte** [comment]
Good! Thanks!
